doc/kb_sdk_impl_methods.md

```diff
@@ -0,0 +1,201 @@
+# <A NAME="top"></A>![alt text](https://avatars2.githubusercontent.com/u/1263946?v=3&s=84 "KBase") [KBase SDK](../README.md)
+
+1. [Install SDK Dependencies](kb_sdk_dependencies.md)
+2. [Install SDK with Docker](kb_sdk_dockerized_install.md)
+3. [Create Module](kb_sdk_create_module.md)
+4. [Edit Module and Method(s)](kb_sdk_edit_module.md)
+5. **Implement Method(s)**
+6. [Specify User Interface](kb_sdk_make_ui.md)
+7. [Locally Test Module and Method(s)](kb_sdk_local_test_module.md)
+8. [Register Module](kb_sdk_register_module.md)
+9. [Test in KBase](kb_sdk_test_in_kbase.md)
+10. [Complete Module Info](kb_sdk_complete_module_info.md)
+11. [Deploy](kb_sdk_deploy.md)
+
+#### <A NAME="impl"></A>5. Edit Impl file
+
+In the lib/\<MyModule\>/ directory, edit the <MyModule>Impl.py (or *.pl) "Implementation" file that defines the methods 
+available in the module.  You can follow this guide for interacting with [KBase Data Types](doc/kb_sdk_data_types.md).  
+Basically, the process consists of obtaining data objects from the KBase workspace, and either operating on them 
+directly in code or writing them to scratch files that the tool you are wrapping will operate on.  Result data should 
+be collected into KBase data objects and stored back in the workspace.
```

Comment:
I can help here if you need it. BTW DFU doesn't 'replace' the workspace - it wraps it and makes things a bit more simple (like automatic provenance handling). 

Where are you seeing direct WS access in the example? https://github.com/kbase/kb_sdk/blob/master/src/java/us/kbase/templates/module_python_impl.vm.properties